### PR TITLE
Added common files for lafaspot

### PR DESCRIPTION
**Pull request by manish211**
Added the below files

new file:   core/src/main/java/com/lafaspot/jmetrics/common/BaseMonitor.java
	new file:   core/src/main/java/com/lafaspot/jmetrics/common/MonitorDirectory.java
	new file:   core/src/main/java/com/lafaspot/jmetrics/common/MonitorStateHandler.java
	new file:   core/src/main/java/com/lafaspot/jmetrics/common/TimeValue.java
	new file:   core/src/main/java/com/lafaspot/jmetrics/common/TypeFormatException.java

### CODE REVIEW

**Comment by pritish5699** [comment]
@manish211 Can we create following packages:

common.monitor: BaseMonitor, MonitorDirectory, MonitorStateHandler
common.datatype: TimeValue, TypeFormatException

File: core/src/main/java/com/lafaspot/jmetrics/common/MonitorManager.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * hostname|coloname|clustername|clusterType to namespace list while creating monitor.
+     *
+     * @param namespace List of names to be used in case monitor object needs to be created.
+     * @return Monitor instance of type T corresponding to namespace defined in passed param.
+     */
+    public T getMonitor(@Nonnull final List<String> namespace) {
+    	final Set<String> orderedNamespace = new TreeSet<>();
+        orderedNamespace.addAll(namespace);
+        orderedNamespace.addAll(constNamespaceSet);
+        final StringJoiner namespaceJoiner = new StringJoiner("|");
```
Comment (pritish5699) [inline core/src/main/java/com/lafaspot/jmetrics/common/MonitorManager.java]: Can we use String.join instead. Sample:
https://reversecoding.net/java-8-convert-list-string-comma/

File: core/src/main/java/com/lafaspot/jmetrics/common/MonitorManager.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        for (String element : orderedNamespace) {
+            namespaceJoiner.add(element);
+        }
+        return monitorDirectory.getMonitor(clazz.getClass().getName() + ":namespace="
```
Comment (pritish5699) [inline core/src/main/java/com/lafaspot/jmetrics/common/MonitorManager.java]: String append is expensive. Use StringBuilder.

**Comment by pritish5699** [comment]
overall looks good. Can you increment version in pom.xml? That will be needed in order to merge the PR. 👍

**Comment by manish211** [comment]
taken further by @joshiaditya1993  in this pull request **https://github.com/lafaspot/jmetrics/pull/21**. Closing this one.

**Pull request closed without merging**